framework/validators/EmailValidator.php

```diff
@@ -176,4 +176,28 @@ public function getClientOptions($model, $attribute)
 
         return $options;
     }
+
+    /**
+     * @param string $value
+     * @return string|bool returns string if it is valid and/or can be converted, bool false if it can't be converted and/or is invalid
+     * @see https://github.com/yiisoft/yii2/issues/18585
+     */
+    private function validateIdnValue ($value)
```

Comment:
It is better to separate value validation and value conversion i.e. `$this->idnToAscii($matches['local']);` should stay as is but then it should be validated.

Reply:
It has already been merged, but I am happy to submit a new pull request with changes.

Not 100% sure I am following you based on where this comment is, but it looks like what you want could be achieved by changing line 80 to:

```
$matches['local'] = $this->idnToAscii($matches['local']); // bring this back
$matches['local'] = $this->validateIdnValue($matches['local']); // this would be the new value for local
```
then removing line 187, as it would be redundant and called outside of this method.

If that is what you are looking for, I can create a new branch and submit it.

Reply:
> It has already been merged, but I am happy to submit a new pull request with changes.

Nope. It was not merged yet. Just push to the same branch and changes will end up in this PR.

Reply:
Checked it again. That's not really validation. Proposed change separately.